### PR TITLE
Weather app

### PR DESCRIPTION
**Pull request by iklaq**
used async and await and fetch api for displaying weather of different cities

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// Initializing all elements constants
+const temperateField = document.querySelector(".weather1");
```
Comment (balwant-jtc) [inline index.js]: temperateField > temperatureField

File: index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    // Calling update Dom Function
+    updateDom(temp_c, name, localtime, icon, text);
+  } catch (error) {
+    alert("Location not found");
+  }
+};
+
+// Function to update Dom
+function updateDom(temperate, city, time, emoji, text) {
```
Comment (balwant-jtc) [inline index.js]: Change `temperate` to `temperature`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  const exactTime = time.split(" ")[1];
```
Comment (balwant-jtc) [inline index.js]: use destructuring for getting `date` and `time` from `time.split()`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+// Adding event listen to the form
+form.addEventListener("submit", search);
+
+// Function to search the location
+function search(e) {
+  e.preventDefault();
+
+  target = searchField.value;
+
```
Comment (balwant-jtc) [inline index.js]: remove this new line

File: index.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+const dateField = document.querySelector(".weather2 span");
+const emojiField = document.querySelector(".weather3 img");
+const weatherField = document.querySelector(".weather3 span");
+const searchField = document.querySelector(".searchField");
+const form = document.querySelector("form");
+
+// Function to fetch Data from Weather API
+const fetchData = async (target) => {
+  try {
+    const url = `https://api.weatherapi.com/v1/current.json?key=5b27a6ef3547402582e62007222306&q=${target}`;
```
Comment (balwant-jtc) [inline index.js]: Generally, such API Url's should be kept in separate file `constants.js`, put this url there and import from `constants.js` to here

**Review by balwant-jtc — CHANGES_REQUESTED** [review]

File: index.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      return "Wednesday";
+
+    case 4:
+      return "Thursday";
+
+    case 5:
+      return "Friday";
+
+    case 6:
+      return "Saturdat";
```
Comment (roopaljasnani) [inline index.js]: nit: Saturday

File: index.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+function search(e) {
+  e.preventDefault();
+
+  target = searchField.value;
+
+  fetchData(target);
+}
+
+// Function to get the name of day
+function getDayFullName(num) {
```
Comment (roopaljasnani) [inline index.js]: Resort to array index-based fetching over switch in this case:

```
const days = ["Sunday", "Monday",....];
if (num > days.length - 1) num = days.length - 1;
return days[num];
```

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  }
+};
+
+// Function to update Dom
+function updateDom(temperature, city, time, emoji, text) {
+  // Destructuring time and date
+  const {
+    [1]: { exactTime },
+    [0]: { exactDate },
+  } = time.split(" ");
```
Comment (roopaljasnani) [inline src/index.js]: Use array destructuring  here pls

```
const [exactDate, exactTime] = time.split(" ");
```

File: src/index.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+form.addEventListener("submit", search);
+// Function to search the location
+function search(e) {
+  e.preventDefault();
+  target = searchField.value;
+  fetchData(target);
+}
+
+// Function to get the name of day
+function getDayFullName(num) {
```
Comment (roopaljasnani) [inline src/index.js:65]: Also, if the objective is simply to get the day name, you can use this built-in alternative

```
new Date(exactDate).toLocaleString('en-IN', {weekday:'long'})
```

File: src/index.js [hunk 10]
```diff
@@ -63,29 +60,8 @@ function search(e) {
...
-      return "Monday";
-
-    case 2:
-      return "Tuesday";
-
-    case 3:
-      return "Wednesday";
+  const days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+  if (num > days.length - 1) num = days.length - 1;
+  return days[num];
```
Comment (roopaljasnani) [inline src/index.js]: Also, if the objective is simply to get the day name, you can use this built-in alternative

`new Date(exactDate).toLocaleString('en-IN', {weekday:'long'});`

**Pull request merged**